src/expr/src/expr/expr_regexp.rs

```diff
@@ -248,3 +252,513 @@ impl Expression for RegexpMatchExpression {
+                                    ))
+                                }
+                            };
+                        }
+                    } else {
+                        flags = placeholder;
+                    }
+                }
+            }
+        }
+
+        // TODO: Any other error handling?
+        if let Some(_other) = children.next() {
+            // There should not any other option after the `flags`
+            bail!("invalid parameters specified in regexp_replace");
+        }
+
+        // Construct the final `RegexpReplaceExpression`
+        let ctx = RegexpContext::new(&pattern, &flags)?;
+
+        // Set the `global_flag` if 'g' is specified
+        let global_flag = !flags.is_empty() && flags.starts_with('g');
+
+        // Construct the regex used to match and replace `\n` expression
+        // Check: https://docs.rs/regex/latest/regex/struct.Captures.html#method.expand
+        let regex = Regex::new(r"\\([1-9])").unwrap();
+
+        // Check if the syntax is correct
+        if !flags.is_empty() && start != -1 && n == -1 {
+            // `start`, `flag` with no `N` specified is an invalid combination
+            bail!("invalid syntax for `regexp_replace`");
+        }
+
+        Ok(Self {
+            source,
+            ctx,
+            replacement,
+            return_type,
+            start,
+            n,
+            global_flag,
+            regex,
+        })
+    }
+}
+
+impl RegexpReplaceExpression {
+    /// Match and replace one row, return the replaced string
+    fn match_row(&self, text: Option<&str>) -> Option<String> {
+        if let Some(text) = text {
+            // The start position to begin the search
+            let start = if self.start != -1 { self.start - 1 } else { 0 };
+
+            if (self.n == -1 && self.global_flag) || self.n == 0 {
+                // `-g` enabled (& `N` is not specified) or `N` is `0`
+                // We need to replace all the occurrence of the matched pattern
+
+                // See if there is capture group or not
+                if self.ctx.0.captures_len() <= 1 {
+                    println!("Path One");
+
+                    // There is no capture groups in the regex
+                    // Just replace all matched patterns after `start`
+                    return Some(
+                        self.ctx
+                            .0
+                            .replace_all(&text[start as usize..], self.replacement.clone())
+                            .into(),
+                    );
+                } else {
+                    println!("Path Two");
+
+                    // Get the replaced string
+                    let replaced = self
+                        .regex
+                        .replace_all(&self.replacement, "$${$1}")
+                        .to_string();
+
+                    // The position to start searching for replacement
+                    let mut search_start = start as usize;
```

Comment:
I just put the `start` parsing to the very beginning as follows, this should then resolve the unicode index problem.
```
/// Match and replace one row, return the replaced string
fn match_row(&self, text: Option<&str>) -> Option<String> {
  if let Some(text) = text {
      // The start position to begin the search
      let start = if let Some(s) = self.start {
          s - 1
      } else {
          0
      };

      // This is because the source text may contain unicode
      let start = match text.char_indices().nth(start as usize) {
          Some((idx, _)) => idx,
          // With no match
          None => return Some(text.into()),
      };
...
```